### PR TITLE
Add command search methods

### PR DESCRIPTION
**Pull request by Cenngo**
This PR adds methods for exposing the command search logic of the Interaction Service

### CODE REVIEW

File: src/Discord.Net.Interactions/InteractionService.cs [hunk 1]
```diff
@@ -588,6 +588,58 @@ private bool RemoveModuleInternal (ModuleInfo moduleInfo)
             return true;
         }
 
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="ISlashCommandInteraction"/>.
+        /// </summary>
+        /// <param name="slashCommandInteraction">Interaction entity to perform the search with.</param>
+        /// <returns>
+        ///     The search result.
```
Comment (csmir) [inline src/Discord.Net.Interactions/InteractionService.cs]: Maybe note what type of search result is returned to stay consistent?

File: src/Discord.Net.Interactions/InteractionService.cs [hunk 2]
```diff
@@ -588,6 +588,58 @@ private bool RemoveModuleInternal (ModuleInfo moduleInfo)
...
+        /// </summary>
+        /// <param name="messageCommandInteraction">Interaction entity to perform the search with.</param>
+        /// <returns>
+        ///     The search result.
+        /// </returns>
+        public SearchResult<ContextCommandInfo> SearchMessageCommand(IMessageCommandInteraction messageCommandInteraction)
+            => _contextCommandMaps[ApplicationCommandType.Message].GetCommand(messageCommandInteraction.Data.Name);
+
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="IAutocompleteInteraction"/>.
```
Comment (csmir) [inline src/Discord.Net.Interactions/InteractionService.cs]: What does 'loaded' imply, maybe change to 'registered' to match the naming of other methods in the service?

File: src/Discord.Net.Interactions/InteractionService.cs [hunk 3]
```diff
@@ -588,6 +588,58 @@ private bool RemoveModuleInternal (ModuleInfo moduleInfo)
...
+        ///     The search result.
+        /// </returns>
+        public SearchResult<SlashCommandInfo> SearchSlashCommand(ISlashCommandInteraction slashCommandInteraction)
+            => _slashCommandMap.GetCommand(slashCommandInteraction.Data.GetCommandKeywords());
+
+        /// <summary>
+        ///     Search the loaded slash commands using a <see cref="IComponentInteraction"/>.
+        /// </summary>
+        /// <param name="componentInteraction">Interaction entity to perform the search with.</param>
+        /// <returns></returns>
```
Comment (csmir) [inline src/Discord.Net.Interactions/InteractionService.cs]: No return comment present?

**Review by csmir — CHANGES_REQUESTED** [review]

**Comment by Cenngo** [comment]
@Rozen4334 made the changes to the inline docs. take a look

**Review by quinchs — APPROVED** [review]

**Comment by quinchs** [comment]
Waiting on Rozens review

**Review by csmir — APPROVED** [review]
lgtm

**Pull request merged**